pe-parser-library/src/parse.cpp

```diff
@@ -1806,6 +1812,105 @@ bool getRelocations(parsed_pe *p) {
   return true;
 }
 
+bool getDebugDir(parsed_pe *p) {
+  data_directory debugDir;
+  if (p->peHeader.nt.OptionalMagic == NT_OPTIONAL_32_MAGIC) {
+    debugDir = p->peHeader.nt.OptionalHeader.DataDirectory[DIR_DEBUG];
+  } else if (p->peHeader.nt.OptionalMagic == NT_OPTIONAL_64_MAGIC) {
+    debugDir = p->peHeader.nt.OptionalHeader64.DataDirectory[DIR_DEBUG];
+  } else {
+    return false;
+  }
+
+  if (debugDir.Size != 0) {
+    section d;
+    VA vaAddr;
+    if (p->peHeader.nt.OptionalMagic == NT_OPTIONAL_32_MAGIC) {
+      vaAddr =
+          debugDir.VirtualAddress + p->peHeader.nt.OptionalHeader.ImageBase;
+    } else if (p->peHeader.nt.OptionalMagic == NT_OPTIONAL_64_MAGIC) {
+      vaAddr =
+          debugDir.VirtualAddress + p->peHeader.nt.OptionalHeader64.ImageBase;
+    } else {
+      return false;
+    }
+
+    auto numOfDebugEnts = debugDir.Size / sizeof(debug_dir_entry);
+
+    //
+    // this will return the rdata section, where the debug directories are
+    //
+    if (!getSecForVA(p->internal->secs, vaAddr, d)) {
+      return false;
+    }
+
+    //
+    // get debug directory from this section
+    //
+    auto rvaofft = static_cast<std::uint32_t>(vaAddr - d.sectionBase);
```

Comment:
Might need to sanity-check `rvaofft` here, since `d.sectionBase` is file-controlled (I think?)

Reply:
What sanity check should I add? I think `getSecForVA` already checks that it's within the section bounds.

Reply:
Whoops, you're right -- I forgot that the section bounds are checked during construction.